code/modules/mob/living/basic/heretic/rust_spirit.dm

```diff
@@ -0,0 +1,81 @@
+/// Pretty simple mob which creates areas of rust and has a rust-creating projectile spell
+/mob/living/basic/heretic_summon/rust_spirit
+	name = "Rust Walker"
+	real_name = "Rusty"
+	desc = "A grinding, clanking construct which leaches life from its surroundings with every armoured step."
+	icon_state = "rust_walker_s"
+	icon_living = "rust_walker_s"
+	maxHealth = 75
+	health = 75
+	melee_damage_lower = 15
+	melee_damage_upper = 20
+	sight = SEE_TURFS
+	speed = 1
+	ai_controller = /datum/ai_controller/basic_controller/rust_walker
+
+/mob/living/basic/heretic_summon/rust_spirit/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/footstep, FOOTSTEP_MOB_RUST)
+	var/datum/action/cooldown/spell/aoe/rust_conversion/small/conversion = new(src)
+	conversion.Grant(src)
+	ai_controller?.set_blackboard_key(BB_GENERIC_ACTION, conversion)
+
+	var/datum/action/cooldown/spell/basic_projectile/rust_wave/short/wave = new(src)
+	wave.Grant(src)
+	ai_controller?.set_blackboard_key(BB_TARGETTED_ACTION, wave)
+
+/mob/living/basic/heretic_summon/rust_spirit/setDir(newdir)
+	. = ..()
+	if(dir & NORTH)
+		icon_state = "rust_walker_n"
+	else if(dir & SOUTH)
+		icon_state = "rust_walker_s"
+
+/mob/living/basic/heretic_summon/rust_spirit/Life(seconds_per_tick = SSMOBS_DT, times_fired)
+	if(stat == DEAD)
+		return ..()
+
+	var/turf/our_turf = get_turf(src)
+	if(HAS_TRAIT(our_turf, TRAIT_RUSTY))
+		adjustBruteLoss(-1.5 * seconds_per_tick, FALSE)
+		adjustFireLoss(-1.5 * seconds_per_tick, TRUE)
```

Comment:
As per #78657 `updatehealth` isn't called if no damage is healed, meaning with the current setup it might drop some if no fire damage is healed
```suggestion
		var/amt_healed = 0
		amt_healed += adjustBruteLoss(-1.5 * seconds_per_tick, FALSE)
		amt_healed += adjustFireLoss(-1.5 * seconds_per_tick, FALSE)
		if(amt_healed)
			updatehealth()
```
But with that being said, we are a simple mob and we do not take fire damage.
This would only matter if our brute and burn `damage_coeff` were different, which it is not. 
We should probably just heal brute. 
```suggestion
		adjustBruteLoss(-3 * seconds_per_tick)
```